Add `firstInterimResponseStart` and `deliveryType` to PerformanceResourceTiming (#28569)
mdn/content: files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md

---
title: "PerformanceResourceTiming: firstInterimResponseStart property"
short-title: firstInterimResponseStart
slug: Web/API/PerformanceResourceTiming/firstInterimResponseStart
page-type: web-api-instance-property
browser-compat: api.PerformanceResourceTiming.firstInterimResponseStart
---

{{APIRef("Performance API")}}

> The **`firstInterimResponseStart`** read-only property returns a {{domxref("DOMHighResTimeStamp","timestamp")}} immediately after the browser receives the first byte of the interim 1xx response (e.g. 100 continue or 103 early hints) from the server.

Reviewer: ```suggestion
The **`firstInterimResponseStart`** read-only property returns a {{domxref("DOMHighResTimeStamp","timestamp")}} immediately after the browser receives the first byte of the interim 1xx response (e.g. 100 Continue or 103 Early Hints) from the server.
```

There is no _end_ property for `firstInterimResponseStart`.

## Value

The `firstInterimResponseStart` property can have the following values:

- A {{domxref("DOMHighResTimeStamp")}} immediately after the browser receives the first interim bytes of the response from the server.
- `0` if the resource was instantaneously retrieved from a cache.
- `0` if the resource is a cross-origin request and no {{HTTPHeader("Timing-Allow-Origin")}} HTTP response header is used.

## Examples

### Measuring request time

> The `firstInterimResponseStart` and {{domxref("PerformanceResourceTiming.requestStart", "requestStart")}} properties can be used to measure how long the request takes to send an interim response.

Reviewer: ```suggestion
The `firstInterimResponseStart` and {{domxref("PerformanceResourceTiming.requestStart", "requestStart")}} properties can be used to measure how long it takes to the browser to receive an interim response after the sending the request.
```

```js
const request = entry.firstInterimResponseStart - entry.requestStart;
```

> Example using a {{domxref("PerformanceObserver")}}, which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. Use the `buffered` option to access entries from before the observer creation.

Reviewer: ```suggestion
The following example uses a {{domxref("PerformanceObserver")}} to notify of new `resource` performance entries as they are recorded in the browser's performance timeline. The `buffered` option is used for accessing entries from before the observer creation.
```
